src/bgw/job.c

```diff
@@ -416,6 +503,17 @@ zero_guc(const char *guc_name)
 				(errcode(ERRCODE_INTERNAL_ERROR), errmsg("could not set \"%s\" guc", guc_name)));
 }
 
+static LOCKTAG job_session_locktag;
+
+static void
+on_shmem_exit_release_job_lock_callback(int code, Datum arg)
+{
+	if (job_session_locktag.locktag_type != 0)
+	{
+		LockRelease(&job_session_locktag, RowShareLock, /* session_lock */ true);
```

Comment:
we're supposed to acquire a regular lock before mucking with the session locks...

Reply:
This whole machinery not needed for advisory lock